docs/core-design.md

```diff
@@ -0,0 +1,354 @@
+
+##### Alternatives
+
+DRA DeviceClass \- one of the considerations was to re-use the DeviceClass object that comes with the DRA. The problem of reusing this object is that this is not strictly a network-specific object, DeviceClass can be used for any resource, e.g. a GPU. This will introduce confusion in integrating it with other Kubernetes functionality related to networking.
+
+#### API
+
+PodNetwork object is described as follows:
+
+```go
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+
+// PodNetwork represents a logical network in Kubernetes Cluster.
+type PodNetwork struct {
+        metav1.TypeMeta   `json:",inline"`
+        metav1.ObjectMeta `json:"metadata,omitempty"`
+
+        Spec   PodNetworkSpec   `json:"spec"`
+        Status PodNetworkStatus `json:"status,omitempty"`
+}
+
+// PodNetworkSpec contains the specifications for podNetwork object
+type PodNetworkSpec struct {
+
+        // Enabled is used to administratively enable/disable a PodNetwork.
+        // When set to false, PodNetwork Ready condition will be set to False.
+        // Defaults to True.
+        //
+        // +kubebuilder:default=true
+        Enabled bool `json:"enabled,"`
+
+        // Provider specifies the provider implementing this PodNetwork.
+        //
+        // +kubebuilder:validation:MinLength=1
+        // +kubebuilder:validation:MaxLength=253
+        // +optional
+        Provider string `json:"provider,omitempty"`
+
+        // Parameters define provider implementation specific configuration.
+        //
+        // +optional
+        Parameters runtime.RawExtension `json:"parameters,omitempty"`
+}
+
+// PodNetworkStatus contains the status information related to the PodNetwork.
+type PodNetworkStatus struct{
+        // Conditions describe the current conditions of the PodNetwork.
+        //
+        // Known condition types are:
+        // * "Ready"
+        //
+        // +optional
+        // +listType=map
+        // +listMapKey=type
+        // +kubebuilder:validation:MaxItems=5
+        // +kubebuilder:default={{type: "Ready", status: "Unknown", reason:"Pending", message:"Waiting for controller", lastTransitionTime: "1970-01-01T00:00:00Z"}}
+        Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+##### Group
+
+This object will be in the `multinetwork.networking.k8s.io` API group. This will be a new group called `multinetwork`.
```

Comment:
if this uses official API I expect to have conformance tests,

Reply:
Definitely, let me add that to the text.